### PR TITLE
Fix tests requiring timezone alignment

### LINKED ISSUE #37045: SQL: testCurrentTimestampYear fails around New Years Eve
This one was tested today (14:59 PM GMT, December 31st 2018), when in the time zone used in the test the clock already turned 2019 (12:59 AM, January 1st 2019).

```
  2> REPRODUCE WITH: gradlew :x-pack:plugin:sql:qa:security:with-ssl:integTestRunner -Dtests.seed=CF0CBFE0B2953F9F -Dtests.class=org.elasticsearch.xpack.sql.qa.security.JdbcSqlSpecIT -Dtests.method="test {datetime.testCurrentTimestampYear}" -Dtests.security.manager=true -Dtests.locale=bg -Dtests.timezone=Pacific/Guam -Dcompiler.java=11 -Druntime.java=8
FAILURE 3.85s | JdbcSqlSpecIT.test {datetime.testCurrentTimestampYear} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Different result for column [result], entry [1] expected:<2019> but was:<2018>
   >    at __randomizedtesting.SeedInfo.seed([CF0CBFE0B2953F9F:4758803A1C695267]:0)
   >    at org.elasticsearch.xpack.sql.qa.security.JdbcSqlSpecIT.datetime.testCurrentTimestampYear(datetime.sql-spec:120)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.JdbcAssert.doAssertResultSetData(JdbcAssert.java:255)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.JdbcAssert.assertResultSetData(JdbcAssert.java:180)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.JdbcAssert.assertResultSets(JdbcAssert.java:83)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.JdbcAssert.assertResultSets(JdbcAssert.java:67)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.JdbcAssert.assertResultSets(JdbcAssert.java:56)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.SpecBaseIntegrationTestCase.assertResults(SpecBaseIntegrationTestCase.java:127)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.SqlSpecTestCase.doTest(SqlSpecTestCase.java:95)
   >    at org.elasticsearch.xpack.sql.qa.jdbc.SpecBaseIntegrationTestCase.test(SpecBaseIntegrationTestCase.java:88)
   >    at java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by bpintea**
This PR will have the test comparing H2 against ES results have the
H2 timezone sent to ES. This ensures that some timezone sensitive tests
whose results otherwise differ, will always work, no matter the
randomised timezone chosen for the test.

Closes #37045

### CODE REVIEW

File: x-pack/plugin/sql/qa/src/main/resources/datetime.sql-spec [hunk 1]
```diff
@@ -119,8 +119,6 @@ dayOfWeekGroupBy
 SELECT DAY_OF_WEEK(birth_date) day, COUNT(*) c FROM test_emp WHERE DAY_OF_WEEK(birth_date) < 6 GROUP BY day ORDER BY DAY_OF_WEEK(birth_date);
 
 // current
```
Comment (matriv) [inline x-pack/plugin/sql/qa/src/main/resources/datetime.sql-spec]: probably move/remove this comment as well?

**Comment by bpintea** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Review by matriv — COMMENTED** [review]
I think this solution doesn't solve the whole issue.
On the change of the minute/hour/day/month/year it can be that one test (H2) is run before the change of the corresponding field and the ES test after the change.

File: x-pack/plugin/sql/qa/src/main/resources/datetime.sql-spec (hunk 1 above)
Comment (matriv) [inline x-pack/plugin/sql/qa/src/main/resources/datetime.sql-spec]: Also move/remove this comment?

**Review by matriv — COMMENTED** [review]
Since this is definitely an improvement, I'd say to proceed and merge it first in master and see how the CI runs will behave. The one that I feel could be flaky is the minute one.

Could you please also add tests for CURRENT_DATE/CURRENT_TIME?

**Comment by costin** [comment]
If using H2 is problematic how about doing this test programmatically?

That said, I'm still confused on why, if this fix is needed, it affects only some datetime tests and not all of them? If the TZ is different, so should be the results.

**Comment by bpintea** [comment]
> Since this is definitely an improvement, I'd say to proceed and merge it first in master and see how the CI runs will behave. The one that I feel could be flaky is the minute one.

Sure. I've also added a guard against running before hour change.

> 
> Could you please also add tests for CURRENT_DATE/CURRENT_TIME?

Added those too.

**Comment by bpintea** [comment]
> That said, I'm still confused on why, if this fix is needed, it affects only some datetime tests and not all of them? If the TZ is different, so should be the results.

I've added some more comments to explain why this happens:
https://github.com/elastic/elasticsearch/blob/12784ee92f3a223e620e050bfc381fd9e85d0ce3/x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/SpecBaseIntegrationTestCase.java#L123

But not all tests require that:
https://github.com/elastic/elasticsearch/blob/12784ee92f3a223e620e050bfc381fd9e85d0ce3/x-pack/plugin/sql/qa/src/main/resources/datetime.sql-spec#L133

**Comment by bpintea** [comment]
@elasticmachine merge upstream

File: x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/SqlSpecTestCase.java [hunk 2]
```diff
@@ -13,15 +13,19 @@
...
 import java.util.List;
 import java.util.Locale;
+import java.util.TimeZone;
 
 /**
  * Tests comparing sql queries executed against our jdbc client
  * with those executed against H2's jdbc client.
  */
 public abstract class SqlSpecTestCase extends SpecBaseIntegrationTestCase {
+    private final int TZSYNC_MINUTE_THREASHOLD = 55;
```
Comment (astefan) [inline x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/SqlSpecTestCase.java]: The name of the constant, I believe, has a typo in it: THREASHOLD vs THRESHOLD.

**Review by astefan — APPROVED** [review]
LGTM. Left one comment.

**Review by matriv — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/SqlSpecTestCase.java (hunk 2 above)
Comment (bpintea) [reply x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/SqlSpecTestCase.java]: thanks, fixed it.

**Pull request merged**